packages/11ty/_includes/components/figure/image/image-sequence.webc

```diff
@@ -0,0 +1,221 @@
+     * @returns
+     */
+    debounce(func, wait = 250, immediate = false) {
+      let timeout = null;
+
+      return function () {
+        const context = this;
+        const args = arguments;
+        const later = function () {
+          timeout = null;
+          if (!immediate) {
+            func.apply(context, args);
+          }
+        };
+        const callNow = immediate && !timeout;
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+        if (callNow) {
+          func.apply(context, args);
+        }
+      };
+    }
+
+    handleSequenceNavigation({ clientX, target }) {
+      const draggingLeft = clientX < this.oldX;
+      const draggingRight = clientX > this.oldX;
+      if (draggingLeft) {
+        this.previousCanvas();
+      } else if (draggingRight) {
+        this.nextCanvas();
+      }
+    }
+
+    handleMouseMove(event) {
+      if (this.oldX !== event.clientX) {
+        this.handleSequenceNavigation(event);
+        this.oldX = event.clientX;
+      }
+    }
+
+    handleMouseDown({ clientX }) {
+      this.oldX = clientX;
+      this.addEventListener('mousemove', this.debouncedMouseMove);
+    }
+
+    handleMouseUp() {
+      this.oldX = 0;
+      this.removeEventListener('mousemove', this.debouncedMouseMove);
+    }
+
+    handleMouseLeave() {
+      this.oldX = 0;
+      this.removeEventListener('mousemove', this.debouncedMouseMove);
+    }
+
+    hideImage(imageElement) {
+      imageElement.classList.remove(this.visibleClassName);
+    }
+
+    nextCanvas() {
+      if (this.index === this.totalCanvases - 1) {
+        this.setSequenceIndex(0);
+      } else {
+        this.setSequenceIndex(this.index + 1);
+      }
+    }
+
+    previousCanvas() {
+      if (this.index === 0) {
+        this.setSequenceIndex(this.totalCanvases - 1);
+      } else {
+        this.setSequenceIndex(this.index - 1);
+      }
+    }
+
+    setSequenceIndex(newIndex) {
+      this.index = newIndex;
+      this.synchronizeSequenceInstances();
+    }
+
+    showImage(imageElement) {
+      imageElement.classList.add(this.visibleClassName);
+    }
+
+    synchronizeSequenceInstances() {
+      clearTimeout(this.updateTimer);
+      this.updateTimer = setTimeout(() => {
+        const sequenceInstances = document.querySelectorAll(`image-sequence[sequence-id="${this.sequenceId}"]`);
+        sequenceInstances.forEach((sequence) => {
+          sequence.setAttribute('index', this.index);
+        });
+      }, 250);
+    }
+
+    updateCurrentSequenceImage(newValue, oldValue) {
+      if (!this.root) return;
+      const newIndex = parseInt(newValue);
+      if (!isNaN(newIndex)) {
+        this.showImage(this.images[newIndex]);
+      }
+      const oldIndex = parseInt(oldValue);
+      if (oldIndex !== newIndex && !isNaN(oldIndex)) {
+        this.hideImage(this.images[oldIndex]);
+      }
+    }
+
+  });
+}
```

Comment:
WebC is very strict about semicolons/no `//` comments/other things! It would be good developer experience to lint these files, since our build process won't catch client-side errors.